src/main/java/org/opentripplanner/model/plan/legreference/LegReferenceType.java

```diff
@@ -3,40 +3,60 @@
 import java.io.IOException;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
-import java.text.ParseException;
+import java.util.Arrays;
+import java.util.Optional;
 
 /**
  * Enum for different types of LegReferences
  */
 enum LegReferenceType {
   SCHEDULED_TRANSIT_LEG_V1(
+    1,
     ScheduledTransitLegReference.class,
-    LegReferenceSerializer::writeScheduledTransitLeg,
-    LegReferenceSerializer::readScheduledTransitLeg
+    LegReferenceSerializer::writeScheduledTransitLegV1,
+    LegReferenceSerializer::readScheduledTransitLegV1
+  ),
+
+  SCHEDULED_TRANSIT_LEG_V2(
+    2,
+    ScheduledTransitLegReference.class,
+    LegReferenceSerializer::writeScheduledTransitLegV2,
+    LegReferenceSerializer::readScheduledTransitLegV2
   );
```

Comment:
The design is a bit awkward. But, we can leave it for now. I would love to make a common TokenSerializer witch encapsulate the read/write operations, version and base64 encoding. This should be used in paging as well. I think I will create a task for it.

Reply:
See issue #5451